### PR TITLE
Support OpenShift Logging 6.x

### PR DESCRIPTION
**Pull request by DebakelOrakel**
The parameter `clusterLogForwarding` is deprecated.
Backward compatibility for `clusterLogForwarding` has been removed.

The parameter `clusterLogging` is deprecated and has been removed.
Migrate collector configuration to `clusterLogForwarder.collector`.

The parameter `namespaceLogForwarderEnabled` is deprecated and has been removed.
Multi LogForwarder is enabled by default, see How-To for migration if it was not enabled.

The parameter `namespaceLogForwarder` is deprecated and has been removed.
If you are using Multi LogForwarder with ProjectSyn, migrate them to `adhoc-configuration`.

The parameters `ignore_alerts` and `patch_alerts` are deprecated.
The component is backwards compatible, but moving the parameters to `alerts.ignore` and `alerts.patch` is highly encouraged.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto-generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [ ] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/alertrules.libsonnet [hunk 1]
```diff
@@ -16,63 +14,26 @@ assert
...
 // array entry wasn't prefixed with `~`.
-local user_ignore_alerts = com.renderArray(params.ignore_alerts);
-
-// Upstream alerts to ignore
 local ignore_alerts = std.set(
   // Add set of upstream alerts that should be ignored from processed value of
   // `params.ignore_alerts`
-  user_ignore_alerts
+  com.renderArray(std.get(params, 'ignore_alerts', []))
+  + com.renderArray(std.get(params, 'ignore_alerts', []))
```
Comment (simu) [inline component/alertrules.libsonnet:24]: One of these should be `alerts.ignore` I assume? Additionally, I'd add a deprecation warning for the deprecated field.

```suggestion
  local old_ignore = if std.objectHas(params, 'ignore_alerts') then
    std.trace('Parameter `ignore_alerts` is deprecated, please migrate your config to `alerts.ignore`.', params.ignore_alerts)
  else
    []; 
  com.renderArray(params.alerts.ignore)
  + com.renderArray(old_ignore)
```

File: component/alertrules.libsonnet [hunk 2]
```diff
@@ -16,63 +14,26 @@ assert
...
 );
 
 // Alert rule patches.
 // Provide partial objects for alert rules that need to be tuned compared to
 // upstream. The keys in this object correspond to the `alert` field of the
 // rule for which the patch is intended.
-local patch_alerts = params.patch_alerts;
+local patch_alerts = params.alerts.patch + std.get(params, 'patch_alerts', {});
```
Comment (simu) [inline component/alertrules.libsonnet:31]: Same here:

```suggestion
local patch_alerts =
  local old_patch = if std.objectHas(params, 'patch_alerts') then
    std.trace("Parameter `patch_alerts` is deprecated, please migrate your config to `alerts.patch`, params.patch_alerts);
  else
    {};
  params.alerts.patch + old_patch;
```

File: component/alertrules.libsonnet [hunk 3]
```diff
@@ -103,7 +64,11 @@ local dropInfoRules =
   };
 
 local prometheus_rules(name, groups, baseURL) = kube._Object('monitoring.coreos.com/v1', 'PrometheusRule', name) {
-  metadata+: {
+  metadata: {
+    labels: {
+      name: name,
+    },
+    name: name,
```
Comment (simu) [inline component/alertrules.libsonnet:71]: This seems redundant since it's identical to what you get from `kube._Object` when using `metadata+`.

**Review by simu — APPROVED** [review]
LGTM overall, see inline comments for some smaller things that I noticed.

File: component/alertrules.libsonnet (hunk 3 above)
Comment (DebakelOrakel) [reply component/alertrules.libsonnet:71]: Yes, but `kube._Object` also inserts an empty `annotations` field, which is not applied with `server-side` strategy, causing ArgoCD to see the manifest as unsynced.

File: component/alertrules.libsonnet (hunk 1 above)
Comment (DebakelOrakel) [reply component/alertrules.libsonnet:24]: yes, right 👍

**Pull request merged**